screens/ChatScreen.tsx

```diff
@@ -1,27 +1,47 @@
-import { StyleSheet, Text, View, Pressable, Modal } from 'react-native';
-import { useState } from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+import { useLayoutEffect, useState, useEffect } from 'react';
+import type { NativeStackScreenProps } from '@react-navigation/native-stack';
+import { useQuery } from '@apollo/client';
+
 import ChatHeader from '../components/messages/ChatHeader';
 import MessagesList from '../components/messages/MessageList';
 import ChatInput from '../components/messages/ChatInput';
-import type { NativeStackScreenProps } from '@react-navigation/native-stack';
-import Icon from '@expo/vector-icons/FontAwesome';
-import { MaterialCommunityIcons } from '@expo/vector-icons';
-import { Colors } from '../constants/styles';
-import { Ionicons } from '@expo/vector-icons';
-import { Entypo } from '@expo/vector-icons';
+import { GET_CONTACTS } from '../graphql/queries/Contact';
+import LoadingPage from '../components/ui/Loading';
 
 type RootStackParamList = {
-  Login: undefined;
   Chat: undefined;
   ChatScreen: undefined;
 };
 
 type Props = NativeStackScreenProps<RootStackParamList, 'ChatScreen'>;
 
 const ChatScreen = ({ navigation }: Props) => {
+  const [userData, setUserData] = useState({ name: '', online: true });
   const [reply, setReply] = useState('');
   const [isLeft, setIsLeft] = useState(false);
-  const [showModal, setshowModal] = useState(false);
+
+  const variables = {
+    filter: { id: '21' },
```

Comment:
```suggestion
    filter: { id: contactId },
```
contactId will be passed as a prop to this component. You will get the contact id by modifying the search query as follows

```
search(filter: $filter, contactOpts: $contactOpts, messageOpts: $messageOpts) {
      contact {
        id
        name
        maskedPhone
      }
    }
```